lib/pure/json.nim

```diff
@@ -938,7 +938,7 @@ when defined(js):
     of "[object Array]": return JArray
     of "[object Object]": return JObject
     of "[object Number]":
-      if isInteger(x):
+      if isInteger(x) and 1.0 / cast[float](x) != -Inf: # preserve -0.0 as float
```

Comment:
Why keep the special case? when 1.0, 2.0 ... is int in JS?
It looks weird.

Reply:
because in js, 1.0 and 1 are un-distinguishable, but 0.0 and -0.0 are not indistinguishable, eg:
1.0 / 0.0 == Inf
1.0 / -0.0 == -Inf
(or see also signbit)

Reply:
yep, could you use `signbit` ?

Reply:
I see. Better than nothing.